llvm/lib/Transforms/Vectorize/VectorCombine.cpp

```diff
@@ -4455,22 +4455,15 @@ bool VectorCombine::foldInsExtVectorToShuffle(Instruction &I) {
   SmallVector<int> Mask(NumDstElts, PoisonMaskElem);
 
   bool NeedExpOrNarrow = NumSrcElts != NumDstElts;
-  bool IsExtIdxInBounds = ExtIdx < NumDstElts;
   bool NeedDstSrcSwap = isa<PoisonValue>(DstVec) && !isa<UndefValue>(SrcVec);
   if (NeedDstSrcSwap) {
```

Comment:
I'm not quite sure why this swap logic is necessary, but thats more of a question for old implementation.

Reply:
I think it's just trying to canonicalize directly to having the poison/undef vector argument as the second operand,